exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/aggregate/HashAggTemplate.java

```diff
@@ -1074,7 +1078,12 @@ private void addBatchHolder(int part, int batchRowCount) {
 
   // These methods are overridden in the generated class when created as plain Java code.
   protected BatchHolder newBatchHolder(int batchRowCount) {
-    return new BatchHolder(batchRowCount);
+    return this.injectMembers(new BatchHolder(batchRowCount));
+  }
+
+  protected BatchHolder injectMembers(BatchHolder batchHolder) {
+    CodeGenMemberInjector.injectMembers(cg, batchHolder, context);
```

Comment:
Is it possible to move this method call to the place, general for other similar classes?

Reply:
The ClassGenerator and FragmentContext is needed in this method for general cases, but not for test classes, see example in `ExampleTemplateWithInner.java`

newBatchHolder is override by generated code can be implemented as follows to remove above method 
`protected BatchHolder newBatchHolder(int batchRowCount) { return CodeGenMemberInjector.injectMembers(cg, new BatchHolder(batchRowCount), context); }`
but the `cg` and `context` params are hard coded, not flexable. I think it is better to make a function call to hide the implementation which may differ in each template.